### PR TITLE
fix: use `vim.keymap.set` for filetype

### PR DESCRIPTION
**Pull request by aarnphm**
Fix a regression from 58b4832

Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

### CODE REVIEW

**Comment by CharlesChiuGit** [comment]
Maybe request review from Jint?

File: lua/core/event.lua [hunk 1]
```diff
@@ -47,7 +47,7 @@ vim.api.nvim_create_autocmd("FileType", {
 	},
 	callback = function(event)
 		vim.bo[event.buf].buflisted = false
-		vim.api.nvim_set_keymap("n", "q", "<CMD>close<CR>", { buffer = event.buf, silent = true })
+		vim.keymap.set("n", "q", "<CMD>close<CR>", { buffer = event.buf, silent = true })
```
Comment (Jint-lzxy) [inline lua/core/event.lua:50]:
```console
:h nvim_buf_set_keymap

                                                       *nvim_buf_set_keymap()*
nvim_buf_set_keymap({buffer}, {mode}, {lhs}, {rhs}, {*opts})
    Sets a buffer-local |mapping| for the given mode.

    Parameters:  
      • {buffer}  Buffer handle, or 0 for current buffer

    See also:  
        |nvim_set_keymap()|
```

Also `alpha` has filetype 'nofile', pressing `q` inside the dashboard will immediately close it / throw error if it is the last window.

Sorry for the rush review, a kind of busy here

**Review by Jint-lzxy — CHANGES_REQUESTED** [review]

File: lua/core/event.lua (hunk 1 above)
Comment (aarnphm) [reply lua/core/event.lua:50]: no worries. This shouldn't be the main focus 😄

Comment (aarnphm) [reply lua/core/event.lua:50]: I also tried nvim_buf_set_keymap but it also raise error.

Comment (aarnphm) [inline lua/core/event.lua:50]:
```suggestion
		vim.api.nvim_buf_set_keymap(event.buf, "n", "q", "<CMD>close<CR>", { silent = true })
```

**Review by Jint-lzxy — APPROVED** [review]

**Pull request merged**